### PR TITLE
Reader: limit the number of gravatars shown in the conversation caterpillar

### PR DESCRIPTION
**Pull request by bluefuton**
This PR has two aims:
- limit the number of gravatars shown in the conversation caterpillar to 10
- on small screens (<660px), limit the number of gravatars to 5

### To test

Try the devdocs example at different screen sizes:

http://calypso.localhost:3000/devdocs/blocks/conversation-caterpillar

screen shot 2017-08-04 at 18 25 02

screen shot 2017-08-04 at 18 26 58

### CODE REVIEW

**Comment by jancavan** [comment]
@bluefuton Some usernames are quite long and cause wrapping. 
screenshot 2017-08-07 12 46 27

In my mind, when we designed this, in instances where the names are really long, we may need to drop one or two more gravs. Maybe flex-wrap will work? I'm not sure, I'll have to poke around with it.

File: client/blocks/conversation-caterpillar/docs/fixtures.js [hunk 1]
```diff
@@ -52,4 +52,95 @@ export const comments = [
 		},
 		date: '2016-04-18T15:23:00+00:00',
 	},
```
Comment (samouri) [inline client/blocks/conversation-caterpillar/docs/fixtures.js]: i'd definitely say we are at a point where it'd be better to find real posts.

that way we can easily get multiple different kinds of examples from wordpress en.blog

**Comment by jancavan** [comment]
@bluefuton In https://github.com/Automattic/wp-calypso/pull/16917/commits/ceb052d5c35e5b55bb984324e665676e1267f59c, I made changes so the text is taken into account, but gravs are being dropped from right instead of from the left. How are first 5 gravs getting hidden with the `conversation-caterpillar__gravatar` class?

File: client/blocks/conversation-caterpillar/docs/fixtures.js (hunk 1 above)
Comment (bluefuton) [inline client/blocks/conversation-caterpillar/docs/fixtures.js]: I see your point, but I pressed on with the fixtures so I could always have posts with certain attributes, e.g. post with no avatar URL to make sure it's correctly omitted.

**Comment by bluefuton** [comment]
> How are first 5 gravs getting hidden with the conversation-caterpillar__gravatar class?

With a bit of magic! If there are more than 5 gravs, I apply the class `is-hidden-on-small-screens` to all of the gravatars before the last 5. For example, with 7 gravs:

( .is-hidden-on-small-screens ) ( .is-hidden-on-small-screens ) ( ) ( ) ( ) ( ) ( )

I then set `.is-hidden-on-small-screens` to be `display: none` on <660px.

Do we need to change the way that works?

**Comment by samouri** [comment]
I see `is-hidden-on-small-screens` showing up in the correct gravatars, but you forgot the `display: none` bit 🤔

**Comment by samouri** [comment]
> but gravs are being dropped from the right instead of from the left

I think thats happening now because the width is set and the gravs are displaying on the next line below it but overflow is set to `hidden` so they don't show up.

**Comment by bluefuton** [comment]
> I see is-hidden-on-small-screens showing up in the correct gravatars, but you forgot the display: none bit

It was there originally :) 

https://github.com/Automattic/wp-calypso/pull/16917/commits/3593e778533085bb3f603e0bb296b14c7dfc9e24#diff-7cb65d2277e2a5a9a6c79f16ed043844R14

**Comment by jancavan** [comment]
> I think thats happening now because the gravs are wrapping to the next line below it but overflow is set to hidden so they don't show up.

Yeah, I did this intentionally, but thought there might be some workaround to have them drop from the left instead of right.

**Comment by jancavan** [comment]
Made changes in: https://github.com/Automattic/wp-calypso/pull/16917/commits/2a8926c5b1e7dbd6204fca4265b7aafc72a4b3dc

At <660px:
Before:
screenshot 2017-08-08 13 36 43

After:
screenshot 2017-08-08 13 39 07

For >660px, I added the fade:
screenshot 2017-08-08 14 24 09

I didn't allow wrapping if >660px to avoid something like this from happening:
screenshot 2017-08-08 14 25 38

**Pull request merged**